front/pages/api/w/[wId]/apps/[aId]/state.ts

```diff
@@ -64,10 +64,25 @@ async function handler(
         break;
       }
 
-      await app.update({
+      const updateParams: {
+        savedSpecification: string;
+        savedConfig: string;
+        savedRun?: string;
+      } = {
         savedSpecification: req.body.specification,
         savedConfig: req.body.config,
-      });
+      };
+
+      if (req.body.run) {
+        if (typeof req.body.run != "string") {
+          res.status(400).end();
```

Comment:
Let's move to the new `apiError` for the entire file :grimacing:

Reply:
Done !